src/main/java/ch/njol/skript/test/utils/TestOfflinePlayer.java

```diff
@@ -0,0 +1,255 @@
+package ch.njol.skript.test.utils;
+
+import ch.njol.skript.test.runner.TestMode;
+import com.destroystokyo.paper.profile.PlayerProfile;
+import com.destroystokyo.paper.profile.ProfileProperty;
+import io.papermc.paper.persistence.PersistentDataContainerView;
+import org.bukkit.*;
+import org.bukkit.entity.EntityType;
+import org.bukkit.entity.Player;
+import org.jetbrains.annotations.NotNull;
+import org.jspecify.annotations.Nullable;
+
+import java.time.Duration;
+import java.time.Instant;
+import java.util.Date;
+import java.util.Map;
+import java.util.UUID;
+
+public class TestOfflinePlayer implements OfflinePlayer {
```

Comment:
some docs would be welcome
just talk about how it's a wrapper for a mostly empty offlineplayer but with a custom profile or smth